src/plugins/data_source/server/client/client_config.ts

```diff
@@ -18,14 +28,70 @@ export function parseClientOptions(
   endpoint: string,
   registeredSchema: any[]
 ): ClientOptions {
+  const sslConfig: DataSourceSSLConfigOptions = {
+    requestCert: true,
+    rejectUnauthorized: true,
+  };
+
+  if (config.ssl) {
+    const verificationMode = config.ssl.verificationMode;
+    switch (verificationMode) {
+      case 'none':
+        sslConfig.rejectUnauthorized = false;
+        break;
+      case 'certificate':
+        sslConfig.rejectUnauthorized = true;
+
+        // by default, NodeJS is checking the server identify
+        sslConfig.checkServerIdentity = () => undefined;
```

Comment:
ok, so the certificate mode would skip the cert validation whether the certificate cert is issued to hostname - https://nodejs.org/api/tls.html#tlscheckserveridentityhostname-cert

Reply:
Marking this as resolved.